situp-core/src/main/java/com/amazon/situp/pipeline/common/PipelineThreadPoolExecutor.java

```diff
@@ -0,0 +1,90 @@
+package com.amazon.situp.pipeline.common;
+
+import com.amazon.situp.pipeline.Pipeline;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The {@link Pipeline} process workers which execute pipeline processors and sinks are spun in n different threads
```

Comment:
`in n` looks like a typo, you might want to capitalize the `N` or reword it to make it more clear.. perhaps something like this?
`The Pipeline process workers that execute pipeline processors and sinks each run in independent threads. If any of these threads fail, have an error, or throw an exception, the pipeline needs to shutdown. We extend the ThreadPoolExecutor to ....`